We used the `fasta_to_variables_table.py` to extract information from protein and nucleotide FASTA files. More specifically, we extracted specific positions per gene of interest into CSV files.

We added that information to a table [link](https://www.github.com/fmenardo/Bgt_fungicides_2024/Dataset/Supplementary_Data_S1.cnv) and performed the rest of the descriptive analyses (`descriptive_stats_haplotypes_cyp51.R` and `fungicide_barplots.R`) and statistics (`fishers_exact_test.R` and `t_test_cyp51_CN.R`).
